src/test/model/processing/test_SingleLogitBiogemeConfig.py

```diff
@@ -45,35 +51,180 @@ def test_b01logit(self):
+                 availability_condition=FunctionalExpression('TRAIN_AV_SP'),
+                 choice_idx=1),
+             'alt2': Alternative(
+                 FunctionalExpression('ASC_SM + B_TIME * SM_TT_SCALED + B_COST * SM_COST_SCALED'),
+                 availability_condition=FunctionalExpression('SM_AV'),
+                 choice_idx=2),
+             'alt3': Alternative(
+                 FunctionalExpression('ASC_CAR + B_TIME * CAR_TT_SCALED + B_COST * CAR_CO_SCALED'),
+                 availability_condition=FunctionalExpression('CAR_AV_SP'),
+                 choice_idx=3)
+         }, choice=FunctionalExpression('CHOICE'))),
+
+        ('b01logit_syntax', SyntaxError,
+         Model(Data(raw_data=__swissmetro_raw_data(), raw_data_path=None,
+                    derivatives={
+                        'SM_COST': FunctionalExpression('SM_CO ****** (GA == 0)'),
+                        'TRAIN_COST': FunctionalExpression('TRAIN_CO * (GA == 0)'),
+                        'CAR_AV_SP': FunctionalExpression('CAR_AV * (SP != 0)'),
+                        'TRAIN_AV_SP': FunctionalExpression('TRAIN_AV * (SP != 0)'),
+                        'TRAIN_TT_SCALED': FunctionalExpression('TRAIN_TT / 100'),
+                        'TRAIN_COST_SCALED': FunctionalExpression('TRAIN_COST / 100'),
+                        'SM_TT_SCALED': FunctionalExpression('SM_TT / 100'),
+                        'SM_COST_SCALED': FunctionalExpression('SM_COST / 100'),
+                        'CAR_TT_SCALED': FunctionalExpression('CAR_TT / 100'),
+                        'CAR_CO_SCALED': FunctionalExpression('CAR_CO / 100'),
+                    }), alternatives={
+             'alt1': Alternative(
+                 FunctionalExpression('ASC_TRAIN + B_TIME * TRAIN_TT_SCALED + B_COST * TRAIN_COST_SCALED'),
+                 availability_condition=FunctionalExpression('TRAIN_AV_SP'),
+                 choice_idx=1),
+             'alt2': Alternative(
+                 FunctionalExpression('ASC_SM + B_TIME * SM_TT_SCALED + B_COST * SM_COST_SCALED'),
+                 availability_condition=FunctionalExpression('SM_AV'),
+                 choice_idx=2),
+             'alt3': Alternative(
+                 FunctionalExpression('ASC_CAR + B_TIME * CAR_TT_SCALED + B_COST * CAR_CO_SCALED'),
+                 availability_condition=FunctionalExpression('CAR_AV_SP'),
+                 choice_idx=3)
+         }, choice=FunctionalExpression('CHOICE'))),
+    ])
+    def test_error_process(self, name: str, expected_exception, model: Model):
+        """
+        Example for using biogeme.
+        Source: https://github.com/michelbierlaire/biogeme/blob/master/examples/swissmetro/b01logit.py (06.07.2023)
+        """
+        config = SingleLogitBiogemeConfig()
+
+        with self.assertRaises(expected_exception):
+            config.process(model)
+
     @staticmethod
     def __example__b01logit() -> Evaluation:
```

Comment:
Hmmm.. I still don't understand the meaning behind this function..

Reply:
I really can't tell...If the function's existence is totally meaningless, than please don't keep it. This will make the code cleaner and shorter. I was confused when I read it, because I didn't see what the function's purpose is...
If you think the function might be helpful in some ways (e.g. can be useful for writing other tests), then keep it with a short comment as explanation., so the reader doesn't get confused like me :)

You can also wait for other reviews. Maybe the others won't experience the same issue as me.